src/lib/wcAutocompleteHandler.ts

```diff
@@ -0,0 +1,39 @@
+import { autocomplete, AutocompleteInstance } from "./autocomplete"
+import type { AutocompleteConfig } from "./config"
+import type { DefaultState } from "../index"
+
+type CustomElement = HTMLElement & { connectedCallback: () => void }
+
+type TemplateProps = {
+  handler: (
+    template: string,
+    options?: object
+  ) => AutocompleteConfig<DefaultState>["render"]
+  defaultTemplate: string
+}
+
+export async function initAutocomplete(
+  element: CustomElement,
```

Comment:
where do you use connectedCallback() ?

Reply:
initAutocomplete is called from within connectedCallback, so you can replace CustomElement with HTMLElement here